Add stft_n_filters to BaseDCUNet

JorisCos opened the pull request:
This PR adds `stft_n_filters` as mentionned in #404.
It also fixes the defaults parameter of DCUNet to match its paper as mentionned in #400

mpariente commented:
There is a problem with CI (1.7.0 build) that might be related to the stride (changed input shape), could someone have a look? 

The other STOI problem on the nightly build _should_ be fixed now.

JorisCos commented:
We can have a closer look to what's wrong with DCCRNet @jonashaag

mpariente commented:
Could you explain the fix please? What was wrong?

JorisCos commented:
Sure, I added `stft_n_filters` in DCUNet and DCCRNet. I also changed the default values of `stft_n_filters` `stft_n_kernel_size` and `stft_stride`. In the tests the models are initialized with the default values and the tests use hard-coded values tied to those default parameters, so they needed to be changed.

JorisCos commented:
At first, I was confused by the jit tests on DCCRNet that kept failing but it was because I didn't changed this ligne 
` masknet_kwargs.setdefault("n_freqs", stft_n_filters // 2)` .

JorisCos commented:
Before, because ` stft_n_filters` was equal to ` stft_kernel_size` ` masknet_kwargs.setdefault("n_freqs", stft_kernel_size // 2)` was ok but it had to be changed

mpariente commented:
Makes complete sense, thanks for the explanation!

mpariente commented on tests/models/models_test.py:
```diff
@@ -155,8 +155,8 @@ def test_dptnet(fb):
 def test_dcunet():
-    _, istft = make_enc_dec("stft", 512, 512)
-    input_samples = istft(torch.zeros((514, 17))).shape[0]
+    _, istft = make_enc_dec("stft", 1024, 1024, 256, 16000)
+    input_samples = istft(torch.zeros((1026, 17))).shape[0]
```
It took time to find out what the bug was because those tests were not clear enough. 
Let's try to improve it: 
- Pass everything to `make_enc_dec` as kwargs, this is unclear currently.
- Add a `n_fft` variable, and reference `n_fft + 2` in the `istft` to be clear.

mpariente commented on tests/models/models_test.py:
```diff
@@ -173,7 +173,7 @@ def test_dcunet():
 
 def test_dccrnet():
-    _, istft = make_enc_dec("stft", 512, 512)
+    _, istft = make_enc_dec("stft", 512, 400, 100, 16000)
```
Same here, name kwargs + introduce `n_fft`.

mpariente reviewed (commented):
After the required edits on the tests, we'll merge that, thanks again

mpariente commented:
Thanks again !
